app/src/main/java/co/nano/nanowallet/NanoUtil.java

```diff
@@ -25,11 +25,14 @@
     public static String generateSeed() {
         int numchars = 64;
         SecureRandom random = SecureRandomUtil.secureRandom();
-        StringBuilder sb = new StringBuilder();
-        while (sb.length() < numchars) {
-            sb.append(Integer.toHexString(random.nextInt()));
+        byte[] randomBytes = new byte[numchars / 2];
+        random.nextBytes(randomBytes);
+      
+        StringBuilder sb = new StringBuilder(numchars);
+        for(byte b: randomBytes){        
+            sb.append(String.format("%02x", b));
```

Comment:
Use `"%02X"` (upper case X) instead and you won't have to convert resulting string to upper case.

Reply:
You're right, fixed!